src/read.c

```diff
@@ -5790,12 +5781,13 @@ static avifResult avifDecoderFindGainMapItem(const avifDecoder * decoder,
         decoder->image->gainMap->image = avifImageCreateEmpty();
         avifImage * image = decoder->image->gainMap->image;
         AVIF_CHECKERR(image, AVIF_RESULT_OUT_OF_MEMORY);
-        if (cicpSet) {
-            image->colorPrimaries = colorPrimaries;
-            image->transferCharacteristics = transferCharacteristics;
-            image->matrixCoefficients = matrixCoefficients;
-            image->yuvRange = yuvRange;
-        }
+        // Look for a colr nclx box. Other colr box types (e.g. ICC) are not supported.
+        AVIF_CHECKRES(avifReadColorNclxProperty(&gainMapItemTmp->properties,
+                                                &image->colorPrimaries,
+                                                &image->transferCharacteristics,
+                                                &image->matrixCoefficients,
+                                                &image->yuvRange,
+                                                NULL));
```

Comment:
Maryla: Please review this PR carefully. I believe it is equivalent to the original code except for how much of `decoder->image->gainMap` may be partially constructed on failure.

I wonder if we should destroy `decoder->image->gainMap` and reset it to NULL on failure (before returning from line 5783 or 5785).